### PR TITLE
OCPBUGS-8266: Downstream Sync 2023-09-19

### PR DESCRIPTION
**Pull request by liornoy**
This PR brings the K8S reporter commit from upstream.
This enables us to run the E2E tests with the `-report` flag and
get a K8S reporter to help us troubleshoot failed tests.

### CODE REVIEW

File: test/util/k8sreporter/reporter.go [hunk 1]
```diff
@@ -1,144 +1,53 @@
...
-		runtimeclient.InNamespace("openshift-sriov-network-operator"))
 
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch network policies: %v\n", err)
-		return
+	dumpNamespace := func(ns string) bool {
+		switch {
+		case ns == namespaces.Test:
+			return true
+		case ns == "openshift-sriov-network-operator":
```
Comment (SchSeba) [inline test/util/k8sreporter/reporter.go:28]: ohhh I miss this one in the u/s review??
there is a variable for the operator namespace please use that one and not this string.
you will need to open the PR u/s first

Comment (liornoy) [reply test/util/k8sreporter/reporter.go:28]: @SchSeba I am looking at the code and don't see a variable that is available for use.
I mean, there is a private const in the controllers package, and a private var in the conformance/tests package.
Do you propose to export the one in the conformance/tests package?

Comment (zeeke) [reply test/util/k8sreporter/reporter.go:28]: I  think @SchSeba is referring to the `OPEATOR_NAMESPACE` environment variable:
It's used in other places in the test suite:

- [test/util/clean/clean.go#L18](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/util/clean/clean.go#L18)
- [test/conformance/tests/init.go#L15](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/conformance/tests/init.go#L15)
- [test/validation/tests/test_validation.go#L28](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/08badf64b9781f772e38ac2eb29315501e93e0b0/test/validation/tests/test_validation.go#L28)

Comment (liornoy) [reply test/util/k8sreporter/reporter.go:28]: Thank you. I filed this: https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/506

Comment (liornoy) [reply test/util/k8sreporter/reporter.go]: The PR upstream merged, I rebased this one.

**Pull request merged**